types/react-data-grid/index.d.ts

```diff
@@ -203,6 +203,16 @@ declare namespace AdazzleReactDataGrid {
                 isSelectedKey?: string;
             }
         }
+        /**
+         * A custom formatter for the select all checkbox cell
+         * @default react-data-grid/src/formatters/SelectAll.js
+         */
+        selectAllRenderer?: React.ReactElement<any> | React.ComponentClass<any> | React.StatelessComponent<any>;
```

Comment:
Taking a look at the implementation of [selectAllRenderer](https://github.com/adazzle/react-data-grid/blob/master/packages/react-data-grid/src/ReactDataGrid.js#L1145), it looks like this isn't allowed to be a ReactElement.